components/domains/steps/checkoutCard.tsx

```diff
@@ -664,12 +669,76 @@ const CheckoutCard: FunctionComponent<CheckoutCardProps> = ({
   const onCurrencySwitch = (type: CurrencyType) => {
     if (type !== CurrencyType.ETH) setLoadingPrice(true);
     setDisplayedCurrency(type);
+    setReducedDuration(0);
+    setReducedDurationToken(null);
+    setBetterReducedDuration(0);
   };
 
   const onUpsellChoice = (enable: boolean) => {
     updateFormState({ isUpselled: enable });
   };
 
+  useEffect(() => {
+    const duration = formState.duration;
+    if (!invalidBalance || !priceInEth) {
+      setReducedDuration(0);
+      setReducedDurationToken(null);
+      setBetterReducedDuration(0);
+      return;
+    }
+    let solutionFoundWithCurrentToken = 0;
+    for (let newDuration = duration - 1; newDuration > 0; newDuration--) {
+      const newPriceInEth = getPriceForDuration(priceInEth, newDuration);
+      let newPrice = newPriceInEth;
+      if (displayedCurrency !== CurrencyType.ETH && quoteData)
+        newPrice = getDomainPriceAltcoin(quoteData.quote, newPriceInEth);
+      const balance = tokenBalances[displayedCurrency];
+      if (!balance) continue;
+      if (BigInt(balance) >= BigInt(newPrice)) {
+        setReducedDuration(newDuration);
+        solutionFoundWithCurrentToken = newDuration;
```

Comment:
you don't `setReducedDurationToken` here as well ? because it's init to `null` on loading

Reply:
Ah I see that you will only update the token currency in the case of `setReducedDurationToken` not being set, right ?

Reply:
In fact in this loop I search for a solution with the current token, so no need to suggest to the user to redirect him to the same token as the one he has already chosen (Because it would show a button "Switch to ETH (7 years)" where ETH is already the current token, so no need to switch anything)